src/chat.rs

```diff
@@ -3506,6 +3582,8 @@ pub async fn remove_contact_from_chat(
                     msg.param.set_cmd(SystemMessage::MemberRemovedFromGroup);
                     msg.param.set(Param::Arg, contact.get_addr());
                     msg.id = send_msg(context, chat_id, &mut msg).await?;
+                } else {
+                    sync = Sync;
```

Comment:
It's on purpose. If we don't send a normal message, then we send a sync message. If it's not supported on another device, that would be just logged. Anyway currently this code is inside
```
if chat.typ == Chattype::Group || chat.typ == Chattype::Broadcast {
```
Yes, `sync` can be made immutable the way you suggested or in other ways, but it would make the code more verbose, i'd better left it as is